<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CustomPalettes</class>
 <widget class="QDialog" name="CustomPalettes">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>515</width>
    <height>354</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>515</width>
    <height>354</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Palette Editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
     <item>
      <layout class="QVBoxLayout" name="paletteListContainer">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>&amp;Palettes:</string>
         </property>
         <property name="buddy">
          <cstring>listPals</cstring>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="listPals"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,1,1">
         <item>
          <widget class="QToolButton" name="cmdAddPal">
           <property name="text">
            <string>&amp;New</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="cmdDelPal">
           <property name="text">
            <string>&amp;Delete</string>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="tbMoreOptions">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>&amp;More </string>
           </property>
           <property name="popupMode">
            <enum>QToolButton::InstantPopup</enum>
           </property>
           <property name="toolButtonStyle">
            <enum>Qt::ToolButtonTextBesideIcon</enum>
           </property>
           <property name="autoRaise">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Colors</string>
       </property>
       <layout class="QVBoxLayout" name="paletteViewContainer">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QListWidget" name="listColors">
              <property name="uniformItemSizes">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="whatsThis">
                 <string>Allows you to edit the selection by typing a new color in hexadecimal #RRGGBB notation.</string>
                </property>
                <property name="text">
                 <string>He&amp;x:</string>
                </property>
                <property name="buddy">
                 <cstring>leColor</cstring>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="leColor">
                <property name="whatsThis">
                 <string>Allows you to edit the selection by typing a new color in hexadecimal #RRGGBB notation.</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="tbEditColor">
                <property name="whatsThis">
                 <string>Displays controls to allow you to choose a new color.</string>
                </property>
                <property name="text">
                 <string>...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,1,0,0,2,0,0">
            <item>
             <widget class="QPushButton" name="cmdAddCol">
              <property name="text">
               <string>&amp;Add Color</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="cmdDelCol">
              <property name="text">
               <string>Re&amp;move Color</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="cmdAddFromList">
              <property name="whatsThis">
               <string>Adds multiple colors from a comma-separated lists.</string>
              </property>
              <property name="text">
               <string>Add from &amp;List</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="cmdWml">
              <property name="whatsThis">
               <string>Generates a sample Wesnoth Markup Language snippet containing a single color palette definition, which you can save to a file or clipboard for later inclusion in your own project.</string>
              </property>
              <property name="text">
               <string>Generate &amp;WML</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="cmdRc">
              <property name="whatsThis">
               <string>Recolors the palette using a given color range.</string>
              </property>
              <property name="text">
               <string>&amp;Recolor</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
  <action name="action_Rename">
   <property name="text">
    <string>&amp;Rename</string>
   </property>
  </action>
  <action name="action_Duplicate">
   <property name="text">
    <string>&amp;Duplicate</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>listPals</tabstop>
  <tabstop>cmdAddPal</tabstop>
  <tabstop>cmdDelPal</tabstop>
  <tabstop>tbMoreOptions</tabstop>
  <tabstop>listColors</tabstop>
  <tabstop>leColor</tabstop>
  <tabstop>tbEditColor</tabstop>
  <tabstop>cmdAddCol</tabstop>
  <tabstop>cmdDelCol</tabstop>
  <tabstop>cmdAddFromList</tabstop>
  <tabstop>cmdWml</tabstop>
  <tabstop>cmdRc</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CustomPalettes</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>252</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CustomPalettes</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>349</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
